src/main/kotlin/domain/Car.kt

```diff
@@ -0,0 +1,43 @@
+package domain
+
+import data.ComparisonResult
+import dto.CarMetadataDTO
+
+class Car(
+    private val generator: NumberGenerator,
+    private val name: String,
+    private var distance: Int = 0
+) {
+
+    fun race(): CarMetadataDTO {
+        val number = generator.generate()
+        if (checkGo(number)) go()
+        return CarMetadataDTO(ComparisonResult.NONE, distance, name)
+    }
+
+    fun compare(winnerDistance: Int): CarMetadataDTO {
+        val difference = distance - winnerDistance
+        val comparisonResult = when {
+            difference > 0 -> ComparisonResult.WIN
+            difference < 0 -> ComparisonResult.LOSE
+            else -> ComparisonResult.DRAW
+        }
+
+        return CarMetadataDTO(comparisonResult, distance, name)
+    }
+
+    private fun checkGo(number: Int): Boolean {
+        if (number in RANGE_LOWER_INCLUSIVE..RANGE_UPPER_INCLUSIVE)
+            return true
+        return false
+    }
```

Comment:
코틀린 문법을 활용하면 아래처럼 작성할 수 있어요 :)
```suggestion
    private fun checkGo(number: Int): Boolean {
        return number in RANGE_LOWER_INCLUSIVE..RANGE_UPPER_INCLUSIVE)
    }
```